apps/teams-test-app/src/components/PagesAPIs.tsx

```diff
@@ -13,9 +14,24 @@ const NavigateCrossDomain = (): React.ReactElement =>
           throw new Error('Target URL is required.');
         }
       },
-      submit: async input => {
-        await pages.navigateCrossDomain(input);
-        return 'Completed';
+      submit: {
+        withPromise: async input => {
+          await pages.navigateCrossDomain(input);
+          return 'Completed';
+        },
+        withCallback: (input, setResult) => {
+          const onComplete = (status: boolean, reason?: string): void => {
+            if (!status) {
+              if (reason) {
```

Comment:
With this flow if `status === false` but there's no `reason` we won't print anything. I think it would be good to add an `else { setResult('status is false but there\'s not reason?! this shouldn\'t happen'); }` kind of code (here and in the other APIs you're updating) to surface this to the user if we ever get into this state.